components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -2156,13 +2220,14 @@ public static OauthTokenIssuer getOAuthTokenIssuerForOAuthApp(OAuthAppDO appDO)
     }
 
     /**
-     * Get Oauth application information
+     * Get Oauth application information. Internally it uses the tenant present in the carbon context.
      *
-     * @param clientId
-     * @return Oauth app information
-     * @throws IdentityOAuth2Exception
-     * @throws InvalidOAuthClientException
+     * @param clientId Client id of the application.
+     * @return Oauth app information.
+     * @throws IdentityOAuth2Exception      Error while retrieving the application.
+     * @throws InvalidOAuthClientException  If an application not found for the given client ID.
      */
+    @Deprecated
```

Comment:
- Add the deprecation note to use the new method. Say why this is deprecated.
- Check other places